Mathlib/CategoryTheory/Sites/Equivalence.lean

```diff
@@ -0,0 +1,231 @@
+def sheafCongr : Sheaf J A ≌ Sheaf (e.locallyCoverDense J).inducedTopology A where
+  functor := e.sheafCongr_functor J A
+  inverse := e.sheafCongr_inverse J A
+  unitIso := e.sheafCongr_unitIso J A
+  counitIso := e.sheafCongr_counitIso J A
+  functor_unitIso_comp X := by
+    ext
+    simp only [id_obj, sheafCongr_functor_obj_val_obj, comp_obj, Sheaf.instCategorySheaf_comp_val,
+      NatTrans.comp_app, sheafCongr_inverse_obj_val_obj, Opposite.unop_op,
+      sheafCongr_functor_map_val_app, sheafCongr_unitIso_hom_app_val_app,
+      sheafCongr_counitIso_hom_app_val_app, sheafCongr_functor_obj_val_map, Quiver.Hom.unop_op,
+      Sheaf.instCategorySheaf_id_val, NatTrans.id_app]
+    simp [← Functor.map_comp, ← op_comp]
+
+variable [HasSheafify (e.locallyCoverDense J).inducedTopology A]
+
+/-- Transport a presheaf to the equivalent category and sheafify there. -/
+noncomputable
+def transportAndSheafify : (Cᵒᵖ ⥤ A) ⥤ Sheaf J A :=
+  e.op.congrLeft.functor ⋙ presheafToSheaf _ _ ⋙ (e.sheafCongr J A).inverse
+
+/-- An auxiliary definition for the sheafification adjunction. -/
+noncomputable
+def transportIsoSheafToPresheaf : (e.sheafCongr J A).functor ⋙
+    sheafToPresheaf (e.locallyCoverDense J).inducedTopology A ⋙
+    e.op.congrLeft.inverse ≅ sheafToPresheaf J A :=
+  NatIso.ofComponents (fun F ↦ isoWhiskerRight e.op.unitIso.symm F.val)
+    (by intros; ext; simp [Equivalence.sheafCongr])
+
+/-- Transporting and sheafifying is left adjoint to taking the underlying presheaf. -/
+noncomputable
+def transportSheafificationAdjunction : transportAndSheafify J e A ⊣ sheafToPresheaf J A :=
+  ((e.op.congrLeft.toAdjunction.comp (sheafificationAdjunction _ _)).comp
+    (e.sheafCongr (A := A) J).symm.toAdjunction).ofNatIsoRight (transportIsoSheafToPresheaf _ _ _)
+
+noncomputable instance : PreservesFiniteLimits <| transportAndSheafify J e A where
+  preservesFiniteLimits _ := compPreservesLimitsOfShape _ _
+
+/-- Transport `HasSheafify` along an equivalence of sites. -/
+theorem hasSheafify : HasSheafify J A :=
+  HasSheafify.mk' J A (transportSheafificationAdjunction J e A)
+
+variable {A : Type*} [Category A] {B : Type*} [Category B] (F : A ⥤ B)
+  [(e.locallyCoverDense J).inducedTopology.HasSheafCompose F]
+
+theorem hasSheafCompose : J.HasSheafCompose F where
+  isSheaf P hP := by
+    let K := (e.locallyCoverDense J).inducedTopology
+    have hP' : Presheaf.IsSheaf K (e.inverse.op ⋙ P ⋙ F) := by
+      change Presheaf.IsSheaf K ((_ ⋙ _) ⋙ _)
+      apply HasSheafCompose.isSheaf
+      exact e.inverse.op_comp_isSheaf K J ⟨P, hP⟩
+    replace hP' : Presheaf.IsSheaf J (e.functor.op ⋙ e.inverse.op ⋙ P ⋙ F) :=
+      e.functor.op_comp_isSheaf _ _ ⟨_, hP'⟩
+    exact (Presheaf.isSheaf_of_iso_iff ((isoWhiskerRight e.op.unitIso.symm (P ⋙ F)))).mp hP'
+
+end Equivalence
+
+variable {C : Type (u+1)} [LargeCategory C] [EssentiallySmall C] (J : GrothendieckTopology C)
+variable (A : Type*) [Category A] --[LargeCategory A]
```

Comment:
The following seems more general to me, and it seems it compiles:
```suggestion
variable {C : Type*} [Category C] [EssentiallySmall C] (J : GrothendieckTopology C)
variable (A : Type*) [Category A]
```
Would you agree? (And does this work for your application?)